spec/predicates/release.md

```diff
@@ -0,0 +1,95 @@
+# Predicate type: Release
+
+Type URI: https://in-toto.io/attestation/release
+
+Version 0.1
+
+## Purpose
+
+To authoritatively link from a specific release name and version string in a registry, to the artifact names and hashes that make up that release.
+
+## Use Cases
+
+When receiving a new release version, package registries can publish a release attestation covering the artifact names and hashes that make up that release. This allows consumers of that release version to ensure the artifacts they are consuming have not been tampered with.
+
+If these release attestations are optionally published to a transparency log, package authors (or other interested parties) can monitor when a new version of a package is released.
+
+These use cases are not hypothetical; both of them are the case today for npm's [build provenance feature], which includes a [publish attestation]. Note that while npm calls this a publish attestation, calling it a release attestation better reflects that it's coming from the package registry. Publishing often refers to an author sending content to the registry, as in [PyPI's trusted publishers feature].
+
+## Prerequisites
+
+This predicate depends on the [in-toto attestation framework], as well as the [purl-spec] for identifying packaging ecosystems.
+
+Perhaps surprisingly, this predicate does not depend on [SLSA Provenance], but they are better together. In an ecosystem where some (but not all) packages have SLSA Provenance and you query by artifact hash, you can't tell the difference between an artifact that has been tampered with and one that does not yet have SLSA provenance. If the ecosystem has release attestations, you can give authoritative answers to what artifacts make up a given release version, and what hashes those artifacts should have.
+
+## Model
+
+This predicate is for the final stages of the software supply chain, where consumers are looking for attestations that the software has not been tampered with during distribution.
+
+## Schema
+
+### Fields
+
+- **purl, required** string (ResourceURI)
+  - A purl uniquely identifying a specific release name and version from a package registry.
+
+### Parsing Rules
+
+The purl field MUST be parsed using the [purl-spec]. It MUST include a version (which is OPTIONAL in the [purl-spec]).
+
+## Example
+
+```json
+{
+  "_type": "https://in-toto.io/Statement/v1",
+  "subject": [
+    {
+      "name": "@scope/my-package",
+      "digest": {
+        "sha256": "deadbeef0123456789..."
+      }
+    }
+  ],
+  "predicateType": "https://in-toto.io/attestation/release/v0.1",
+  "predicate": {
+    "purl": "pkg:npm/@scope/my-package@1.2.3",
+  }
+}
+```
+
+If a release has multiple artifacts that might be consumed separately, the attestation SHOULD have a subject per artifact:
+
+```json
+{
+  "_type": "https://in-toto.io/Statement/v1",
+  "subject": [
+    {
+      "name": "urllib3-2.1.0.tar.gz",
```

Comment:
I might also recommend a file size here to prevent [endless data attacks](https://theupdateframework.github.io/specification/v1.0.33/index.html#goals-to-protect-against-specific-attacks).

Reply:
This is a really interesting suggestion!

Technically this is part of the [statement layer](https://github.com/in-toto/attestation/blob/main/spec/v1/statement.md#statement-layer-specification) (specifically the [resource description](https://github.com/in-toto/attestation/blob/main/spec/v1/resource_descriptor.md)) and not the proposed release attestation predicate. Maybe an in-toto maintainer can help us figure out how to propose this change to the resource description (or if it could just be part of the existing `annotations`)?

However, I think statement layer proposals are probably separate from predicate proposals, even if they inform one another.

Reply:
interesting point! My initial reaction is to suggest using the `annotations` field for this sort of extra metadata, because file size is a resource-specific piece of info. But I can also see the benefits of adding an optional `size` field to the object. The parsing rules for this field may change depending on the type of resource being described, so that may get a little tricky. 

Do feel free to open an issue on this, because yes, changing the RD spec should be handled separately from this PR.

Reply:
It's also the sort of stuff that TUF was designed for, so maybe it's not necessary here, but it's something worth thinking about upstream with the RD spec.

Reply:
We have determined this is out of scope for this ITE-9 process.